### PR TITLE
XW-2585 | Add File page handler to Mercury API

### PR DESCRIPTION
**Pull request by idradm**
https://wikia-inc.atlassian.net/browse/XW-2585

### CODE REVIEW

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 1]
```diff
@@ -376,6 +376,19 @@ public function getPage() {
...
+					case NS_FILE:
+						$data['nsSpecificContent'] = MercuryApiFilePageHandler::getFileContent( $title );
+						$data['details'] = MercuryApiArticleHandler::getArticleDetails( $article );
+
+						if ( MercuryApiCategoryHandler::hasArticle( $this->request, $article ) ) {
+							$data['article'] = MercuryApiArticleHandler::getArticleJson( $this->request, $article );
+
+							// Remove namespace prefix from displayTitle, so it can be consistent with title
+							// Prefix shows only if page doesn't have {{DISPLAYTITLE:title} in it's markup
+							$data['article']['displayTitle'] = Title::newFromText($data['article']['displayTitle'])->getText();
```
Comment (rogatty) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: Spaces are missing.

**Review by rogatty — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

**Review by rybmat — APPROVED** [review]

File: extensions/wikia/MercuryApi/MercuryApiController.class.php [hunk 2]
```diff
@@ -334,64 +333,60 @@ public function getPage() {
...
+				throw new NotFoundApiException( 'Page doesn\'t exist' );
+			}
+
 			// getPage is cached (see the bottom of the method body) so there is no need for additional caching here
 			$article = Article::newFromID( $title->getArticleId() );
-			$articleExists = $article instanceof Article;
 
-			if ( $articleExists && $title->isRedirect() ) {
+			if ( $title->isRedirect() ) {
 				list( $title, $article, $data ) = $this->handleRedirect( $title, $article, $data );
```
Comment (idradm) [inline extensions/wikia/MercuryApi/MercuryApiController.class.php]: Redirect to deleted page will result in correct api response with `redirectEmptyTarget` flag set. Is this behavior correct? Are we handling it on mercury side correctly?

**Comment by rybmat** [comment]
what about improving api-tests for this API?

File: extensions/wikia/MercuryApi/MercuryApiController.class.php (hunk 2 above)
Comment (kvas-damian) [reply extensions/wikia/MercuryApi/MercuryApiController.class.php]: We have the same functionality on prod now. I think it's ok.

**Review by rybmat — APPROVED** [review]

**Review by kvas-damian — APPROVED** [review]

**Pull request merged**